### PR TITLE
jk -- Add DeleteFolder Call and unit Tests

### PR DESCRIPTION
**Pull request by berkeleysquare**

### CODE REVIEW

File: Ds3/Calls/DeleteFolderRequest.cs [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+{
+    public class DeleteFolderRequest : Ds3Request
+    {
+        private string _bucketId;
+        public string BucketId
+        {
+            get { return _bucketId; }
+            set { WithBucketId(value); }
+        }
+        protected DeleteFolderRequest WithBucketId(string bucket)
```
Comment (rpmoore) [inline Ds3/Calls/DeleteFolderRequest.cs]: In other sdks, when we have a required arg, we don't provide a mutator for it.  See https://github.com/SpectraLogic/ds3_net_sdk/blob/master/Ds3/Calls/GetJobRequest.cs for an example of what I mean

Comment (berkeleysquare) [inline Ds3/Calls/DeleteFolderRequest.cs]: Gotcha – do I do another pull request?

jk

Comment (rpmoore) [inline Ds3/Calls/DeleteFolderRequest.cs]: Nope.  Pushing to your repo will update the pull request.
